projects/laji-ui/src/lib/modal/modal/modal.component.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @angular-eslint/no-output-on-prefix */
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, HostListener, Inject, Input, OnDestroy, Output, Renderer2,
+  ViewChild } from '@angular/core';
+import { filter } from 'rxjs/operators';
+
+export type ModalSize = 'sm' | 'md' | 'lg' | 'xl';
+
+@Component({
+  selector: 'lu-modal',
+  template:`
+    <div class="lu-backdrop"></div>
+    <div class="lu-modal-container" [class]="'lu-modal-' + _size" *ngIf="isShown" #container>
+      <lu-button-round *ngIf="!noClose" (click)="hide()" role="neutral" class="lu-modal-close-button">
+        <lu-icon type="close"></lu-icon>
+      </lu-button-round>
+      <div [class]="['lu-modal-content', contentClass]" role="dialog">
+        <ng-content></ng-content>
+      </div>
+    </div>
+  `,
+  styleUrls: ['./modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ModalComponent implements OnDestroy {
+
+  protected _size: ModalSize = 'md';
+
+  /**
+   * One of 'sm', 'md', 'lg', 'xl'. Defaults to 'md'
+   */
+  @Input() set size(size: ModalSize) {
+    this._size = size;
+  }
+
+  // null becuase undefined in the template [class] causes error.
+  @Input() contentClass: string | null = null;
+
+  /**
+   * Modal can't be closed by clicking backdrop and the close button is hidden.
+   * The parent component must handle closing the modal manually.
+   */
+  @Input() noClose = false;
+
+  /**
+   * Emits true/false according to when the modal shows/hides.
+   */
+  @Output() onShownChange = new EventEmitter<boolean>();
+  @Output() onHide = this.onShownChange.pipe(filter(shown => shown === false));
+  @Output() onShow = this.onShownChange.pipe(filter(shown => shown === true));
+  public isShown = false;
```

Comment:
```
$ grep 'public ' -r projects/laji/**/*.ts | wc -l
554
```

I don't think we have any convention on this... They add a bit syntactic noise yes. On the other hand, they do provide some meta semantic purpose, as they relay information about the intent of the stated variables. "This variable is designed to be used by the template". "This method is deliberately designed for external calls and will work. It's not a private method and calling it won't break the component". If a variable/method doesn't have `public` stated, you can't know whether the developer has forgotten to set it `private`, of if it's really safe to use.

The same semantics could be achieved by being very systematic about `private` modifier, leaving no modifier for public. That would be cool but yeah we don't have any convention across laji-repo so not really an option.

Reply:
Sure there are a lot of publics out there, but vast majority of public vars/methods are not explicitly marked as public.

There's two somewhat reasonable options imo:
* Either all members are explicitly marked https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-member-accessibility.md
* Or explicit publics are disallowed

The former adds a lot of redundancy, but makes sure that people don't forget about member accessibility concerns. The latter is way more convenient and easier to read (imo), but less safe. I think that it should be reasonable to expect a coder to already consider member accessibility whenever they are creating declarations without requiring to be explicit. If they aren't considering it, then PR review should rectify these issues. So I'm in favor of the latter.

Mixed usage of explicit publics and implicit publics looks pretty ugly / makes code harder to read, at least for me. It's basically totally arbitrary whether the dev chooses to be explicit or not.

Reply:
I'm ok with banning `public` modifier but I'd rather have a lint rule for that. Adding such rule shouldn't be too hard, the linter should be automatically able to remove the existing `public`s. Otherwise issues like these are arbitrarily resolved according to who happens to do the reviewing. I'm just gonna remove the modifiers and not add the linter here, because we're creeping hard already :)